### PR TITLE
Updated Nginx & HTTPD config

### PR DESCRIPTION
**Pull request by dmikusa-pivotal**
- stripped unnecessary logic for folder redirects from nginx.conf
- added option to nginx.conf and httpd.conf to redirect HTTP to HTTPS
   - this redirect happens only when behind a proxy (i.e. x-forwarded-* headers are set) and only when the proto is not set to `https`

https://www.pivotaltracker.com/story/show/166590861

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -100,14 +102,15 @@ type BuildpackYAML struct {
...
-	ServerAdmin  string    `yaml:"serveradmin"`
-	Redis        Redis     `yaml:"redis"`
-	Memcached    Memcached `yaml:"memcached"`
+	Version             string    `yaml:"version"`
+	WebServer           string    `yaml:"webserver"`
+	WebDirectory        string    `yaml:"webdirectory"`
+	LibDirectory        string    `yaml:"libdirectory"`
+	Script              string    `yaml:"script"`
+	ServerAdmin         string    `yaml:"serveradmin"`
+	EnableHTTPSRedirect bool      `yaml:"disable_https_redirect"`
```
Comment (ryanmoran) [inline config/config.go:111]: The name of this field seems to contradict its `yaml` tag. Is that by design?

Comment (dmikusa-pivotal) [reply config/config.go]: No, that's a typo. Sorry. Good catch. I pushed an update and fixed this plus added a test to validate. Let me know if it's easier to resubmit this PR, it looks like there are merge conflicts.

**Comment by ryanmoran** [comment]
This looks good, but we'll need to resolve these conflicts before we can merge it.

**Comment by dmikusa-pivotal** [comment]
OK, just to be clear. Would you like me to update the PR or are you going to resolve the conflicts? It seems like it's just import name changes to be updated. Nothing too crazy.

**Comment by ryanmoran** [comment]
I see now how I wasn't very clear. I'll go ahead and just resolve the conflicts myself. Thanks!

**Pull request merged**